btech/btfi/Codec.cc

```diff
@@ -1130,12 +1130,6 @@ Encoder::writeNonEmptyOctets_len_bit2(FI_PInt32 len)
 	assert(getBitOffset() == 1); // C.22.2
 	assert(len <= FI_PINT32_MAX);
 
-	if (len > FI_UINT_TO_PINT(FI_LENGTH_MAX)) {
```

Comment:
This condition simply cannot happen. `len` is a 32 bit integer. `FI_UINT_TO_PINT(FI_LENGTH_MAX)` on a 32 bit platform will always larger or equal. On a 64 bit `len` simply will always be less.  Hence it seemed sensible to remove this check.